### DISCUSSION TITLE
Has Renovate OSS ever been professionally pentested?

### DISCUSSION THREAD

**Discussion by jthvai** [renovatebot/renovate#27275] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

I completely understand if this is confidential.

Following on from a discussion [on Codeberg](https://codeberg.org/Codeberg/Contributing/issues/47#issuecomment-1560620) where @viceice weighed in and pointed out a possible security hole in Renovate OSS;

I understand Mend is [primarily an appsec company](https://www.mend.io/about-us/), and I'm under the impression that Renovate OSS is the core of Renovate CE and EE - which would seem to suggest that there are both financial incentives and means to have Renovate OSS professionally pentested. So, has it ever been done?

**Comment by HonkingGoose** [comment]
Did you read the [Security and permissions section of the Renovate docs](https://docs.renovatebot.com/security-and-permissions/)? That explains a bit more about Renovate's security stance. It does not mention any pentesting though.

I'll let the maintainers respond to your question. 😉

**Comment by rarkins** [comment, marked as answer]
The Renovate Open Source project has not been pen tested directly. Mend.io is not a pen testing company itself so our own infrastructure is checked periodically by third parties like any other

**Reply by jthvai** [reply]
Thanks!
